src/Adapter/MSTest.TestAdapter/Execution/TestAssemblyInfo.cs

```diff
@@ -208,33 +213,46 @@ public void RunAssemblyInitialize(TestContext testContext)
             return null;
         }
 
+        Exception? assemblyCleanupException = null;
         lock (_assemblyInfoExecuteSyncObject)
         {
             try
             {
-                AssemblyCleanupMethod.InvokeAsSynchronousTask(null);
-
-                return null;
+                assemblyCleanupException = MethodRunner.RunWithTimeoutAndCancellation(
+                     () => AssemblyCleanupMethod.InvokeAsSynchronousTask(null),
+                     new CancellationTokenSource(),
+                     AssemblyCleanupMethodTimeoutMilliseconds,
+                     AssemblyCleanupMethod,
+                     Resource.AssemblyCleanupWasCancelled,
+                     Resource.AssemblyCleanupTimedOut);
             }
             catch (Exception ex)
             {
-                var realException = ex.GetRealException();
```

Comment:
I think that this one should be preserved right?

You've changed with https://github.com/microsoft/testfx/pull/2570/files#diff-2f90496895f16574f24e69119cf4002a51f31248fa2e1535c57495f4bb725837R241

Reply:
no I have changed it with that: https://github.com/microsoft/testfx/pull/2570/files#diff-2f90496895f16574f24e69119cf4002a51f31248fa2e1535c57495f4bb725837R231

so I don't need to preserve it